tests/TestCase/Database/TypeTest.php

```diff
@@ -251,4 +254,33 @@ public function testSet()
         Type::set('random', $instance);
         $this->assertSame($instance, Type::build('random'));
     }
+
+    /**
+     * @return void
+     */
+    public function testDebugInfo()
+    {
+        $type = new Type('foo');
+        Type::clear();
+        Type::map('bool', BoolType::class);
+        Type::map('int', IntegerType::class);
+        $uuidType = new UuidType('uuid');
+        Type::map('uuid', $uuidType);
+        Type::build('bool');
+
+        $result = $type->__debugInfo();
```

Comment:
Why does each type instance need access to the type registry in its debug output?

Reply:
Currently it mainly also contained

	[protected] _basicTypes => [
		'string' => [
			'callback' => [
				(int) 0 => '\Cake\Database\Type',
				(int) 1 => 'strval'
			]
		],
		'text' => [
			'callback' => [
				(int) 0 => '\Cake\Database\Type',
				(int) 1 => 'strval'
			]
		],
		'boolean' => [
			'callback' => [
				(int) 0 => '\Cake\Database\Type',
				(int) 1 => 'boolval'
			],
			'pdo' => (int) 5
		]
	]

So this cuts it down a bit.
But if you think that should also be omitted, we can further remove.

Reply:
I don't think the instances need to expose that static information. We'll be splitting up the static and instance sides of Types in 4.0